packages/next-tinacms-markdown/src/use-markdown-form.ts

```diff
@@ -0,0 +1,128 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import { useCallback, useEffect, useState, useMemo } from 'react'
+const matter = require('gray-matter')
+import * as yaml from 'js-yaml'
+
+import {
+  useWatchFormValues,
+  useForm,
+  useCMS,
+  FormOptions,
+  Field,
+} from 'tinacms'
+
+/**
+ * A datastructure representing a MarkdownFile stored in Git
+ */
+
+export interface MarkdownFile  {
+  fileRelativePath: string
+  frontmatter: any
+  markdownBody: string
+}
+
+export interface Options {
+  id?: string
+  label?: string
+  fields?: Field[]
+  actions?: FormOptions<any>['actions']
+}
+
+export function toMarkdownString(markdownFile: MarkdownFile) {
+  return (
+    '---\n' +
+    yaml.dump(markdownFile.frontmatter) +
+    '---\n' +
+    (markdownFile.markdownBody || '')
+  )
+}
+/**
+ * Creates a TinaCMS Form for editing a MarkdownFile in Git
+ */
+export function useMarkdownForm (
+  markdownFile: MarkdownFile,
+  options: Options = {}
+) {
+  const cms = useCMS()
+
+  const [valuesInGit, setValuesInGit] = useState()
+
+  const valuesOnDisk = useMemo(
```

Comment:
I vaguely think the `useMemo` is unnecessary here. That callback isn't doing any processing, just creating a new object. I think `const valuesOnDisk = markdownFile` would be fine. 

I like making the new variable just for the sake of making it easier to understand.

I could be wrong on this. We could do a quick check to see if it causes a bunch of unnecessary rendering without the `useMemo`

Reply:
I was having a hard time getting it to work without `useMemo` but I will do some more testing

Reply:
Okay cool! If that's the case then I would just add `markdownFile.fileRelativePath` to the dependency array